components/theia/packages/gitpod-extension/src/browser/gitpod-user-storage-provider.ts

```diff
@@ -68,6 +73,16 @@ export class GitpodUserStorageProvider implements FileSystemProviderWithFileRead
     }
 
     async readFile(resource: URI): Promise<Uint8Array> {
+        if (this.delegate && this.delegate.readFile) {
```

Comment:
Should no we await both and take one which is fresher?

Reply:
> or we alway write to the file and prefer it if it exists?

that's the intention

Reply:
then we should remove `value.length > GitpodUserStorageProvider.MAX_NUM_BYTES` guard from `writeFile` to make sure that data written to the file and database